### ISSUE TITLE
Inserting a drawing / chart causes print options to break in version 1.3.1

### ISSUE THREAD

**Issue by rf1234** [PHPOffice/PhpSpreadsheet#610]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
In version 1.2.1 there was no issue with print options: I don't set any and the template I use is issue checked so that there shouldn't be any problems.

### What is the current behavior?
In version 1.3.1 I get an issue with Print Options. Excel repairs the file and removes print options.

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$spreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();

// add code that show the issue here...
```

**Comment by fyrye** [comment]
@rf1234 what template are you using? 
How do we reproduce the issue you are experiencing?
What do you mean by `issue checked so that there shouldn't be any problems.`?

**Comment by rf1234** [comment]
I guess I found the problem now: If I insert a drawing into one of the sheets (it can be a jpeg or an Excel chart) the problem occurs:

This rather simple code which does almost nothing causes the print option problem:
```
$templateFile = "Darlehensuebersicht_template.xlsx";
    //we have xlsx and xls input files the input file types are hence Xlsx or Xls
    $readerType = ucfirst( substr($templateFile, strrpos($templateFile, '.') + 1) );
    if ( $readerType === 'Xlsm' ) {
        $readerType = 'Xlsx';
    }
    
    //open spreadsheet template
    $templateFilePath = __DIR__ . '/reporting/templates/' . $templateFile;
    
    $reader = IOFactory::createReader($readerType);
    $reader->setIncludeCharts(false); //we don't want to load charts
    $spreadsheet = $reader->load($templateFilePath);
    
    $outputFile = "Darlehensuebersicht.xlsx";
    
    $outputFileType = substr($outputFile, strrpos($outputFile, '.') + 1);
    $writerType = ucfirst( $outputFileType );
    if ( $writerType === 'Xlsm' ) {
        $writerType = 'Xlsx';
    }

    //change $outputFile by removing xls or xlsx
    $outputFile = substr($outputFile, 0, strrpos( $outputFile, '.') );
    $reportId = "1.3.1";
    
    $outputFileName = $outputFile . '_' . $reportId . '.' . $outputFileType;
    //set fields to update the file table and to write the file
    $webPath = '/lgfuploads/reporting/' . $outputFileName;
    $systemPath = $_SERVER['DOCUMENT_ROOT'] . $webPath;
    
    $spreadsheet->setActiveSheetIndex(0);
   
    $col = 'K';
    $ws = $spreadsheet->getActiveSheet(); 
    $logoPath = __DIR__ . '/reporting/templates/SmallLogo.jpg';
    $drawing = new Drawing();
    $drawing->setPath($logoPath);
    $drawing->setCoordinates($col.'2');
    $drawing->setHeight(45); //pixels
    $colWidth = $ws->getColumnDimension($col)->getWidth();
    if ($colWidth == -1) { //not defined which means we have the standard width
        $colWidthPixels = 64; //pixels, this is the standard width of an Excel cell in pixels = 9.140625 char units outer size
    } else {                  //innner width is 8.43 char units
        $colWidthPixels = $colWidth * 7.0017094; //colwidht in Char Units * Pixels per CharUnit
    }
    $offsetX = $colWidthPixels - $drawing->getWidth(); //pixels
    $drawing->setOffsetX($offsetX); //pixels
    $drawing->setWorksheet($ws);
    
//write Excel file based on template
    $writer = IOFactory::createWriter($spreadsheet, $writerType);    
    $writer->setIncludeCharts(true);
    $writer->setPreCalculateFormulas(true);
    $writer->save($systemPath);
```
"Issue checked" means I open "file/check for issues" and have Excel fix any issues in the template.

If I remove this code I have no problem:

```
$col = 'K';
    $ws = $spreadsheet->getActiveSheet(); 
    $logoPath = __DIR__ . '/reporting/templates/SmallLogo.jpg';
    $drawing = new Drawing();
    $drawing->setPath($logoPath);
    $drawing->setCoordinates($col.'2');
    $drawing->setHeight(45); //pixels
    $colWidth = $ws->getColumnDimension($col)->getWidth();
    if ($colWidth == -1) { //not defined which means we have the standard width
        $colWidthPixels = 64; //pixels, this is the standard width of an Excel cell in pixels = 9.140625 char units outer size
    } else {                  //innner width is 8.43 char units
        $colWidthPixels = $colWidth * 7.0017094; //colwidht in Char Units * Pixels per CharUnit
    }
    $offsetX = $colWidthPixels - $drawing->getWidth(); //pixels
    $drawing->setOffsetX($offsetX); //pixels
    $drawing->setWorksheet($ws);
    
```
All of this used to work perfectly with version 1.2.1! Hence I moved back to 1.2.1.

I attached four files: the template file, the resulting file without drawing which is ok, the resulting file with drawing which is broken and the picture file that I insert as a drawing. Please help!
[Darlehensuebersicht_1.3.1_without_drawing.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/2222483/Darlehensuebersicht_1.3.1_without_drawing.xlsx)
[Darlehensuebersicht_1.3.1_with_drawing.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/2222484/Darlehensuebersicht_1.3.1_with_drawing.xlsx)
[Darlehensuebersicht_Template.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/2222489/Darlehensuebersicht_Template.xlsx)
![smalllogo](https://user-images.githubusercontent.com/23379336/43123578-e5df50c4-8f24-11e8-8c23-e5a04728f18f.jpg)

**Comment by PowerKiKi** [comment]
@rf1234 are you able to create a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve) of code that exhibits this issue **without relying on an external Excel file** ?

**Comment by rf1234** [comment]
Sorry, no I can't. I never use PHPSpreadsheet without a template. Wouldn't know any reason for using it without a template to be honest. You can find my template attached above and also the full code I used.

**Comment by rf1234** [comment]
@PowerKiKi 
can you help me with this please. I know it seems to work without using a template. But that doesn't help because many people will need to use a template. And according to my experience it works with no template at all. I tried with a lot of them. Made sure they are all checked for integrity etc. But doesn't help. Just tried with version 1.4.0. Same issue with the print options.

**Comment by PowerKiKi** [comment]
To maximise the chance of getting help, you should submit the **minimal** code to reproduce the issue. What you submitted so far is clearly not the minimum. It should be a only a few line of code. You should be able to create a brand new spreadsheet, set printing options, add a drawing and save the file to demonstrate the bug.

**Comment by rf1234** [comment]
@JParkerCFI , you are right. All of the issues I have brought up so far have become "stale" after a while. Virtually nothing got resolved! I am deeply frustrated that I obviously just wasted my time.

**Comment by PowerKiKi** [comment]
This project **heavily** relies on community contributions, because we don't have the resources to do it ourselves. This unfortunately means that if a bug is not resolved in a timely fashion, then your best course of action is to debug it yourself and ideally share back the fix you found.

I understand this can be frustrating, especially for people who are less experienced. But there really is nothing else we can do. For the record we are only 2 "active" people on this project and only in our spare time. I'll do my best to merge incoming PRs, but I personally don't have enough time to spend a whole day debugging a single issue.

**Comment by ger82** [comment]
Hello, in which branch can I commit fixes?

**Comment by rf1234** [comment]
Pull requests!

"First time contributing to PHPOffice/PhpSpreadsheet?
If you know how to fix an issue, consider opening a pull request for it.

You can read this repository’s contributing guidelines or GitHub Help to learn how to open a good pull request."

**Comment by PowerKiKi** [comment]
Pull requests should be made against `master` branch.

**Comment by nicolacardi** [comment]
Hi.
I have a similar problem: I am using the most recent version of PHPspreadsheet (donwloaded yesterday). When I "clone" a worksheet that includes a drawing (I have a logo on each worksheet 
 ) the response is an "out of memory".
Then as a workaround I decided to make a template **without** logo and use the 			
$drawing = new PhpOffice\PhpSpreadsheet\Worksheet\Drawing();
$drawing->setPath(__DIR__ . '/assets/img/logodefViola.png'); (this is not shown correctly as the GitHub editor seems to escape underscore characters)
$drawing->setHeight(40);
$drawing->setCoordinates('B1');  
$drawing->setWorksheet($spreadsheet->getActiveSheet()); to include the logo dinamically everytime I clone a worksheet. This seems to work **BUT**, probably linked to this, I have another issue which might be a character coding issue but can't find how to solve it: when I generate my excel file on localhost it says there is unreadable text in the file (I have tried the exit() instruction at the end of the file, I have tried to remove the '?>' chars, as I found in another thread...still this is the error I see) and then another mask appears where it says PRINTER OPTIONS are REMOVED, which is what I don't want: I need print options (on A3 paper and front/rear ) to be saved and replicated on all worksheets I generate, not removed.... I tried to publish online ad happily I have no more  "unreadable character" message BUT printer options are removed anyway....I guess we are talking about the same story...what do you suggest me to try? Find attached screenshot: they're in italian I make some translation for you : )
<img width="904" alt="screenshot 2019-02-05 16 56 03" src="https://user-images.githubusercontent.com/16313205/52285612-f710ff00-2966-11e9-8f4e-2ba33806bb1d.png"> (translation: ...".xlsx includes unreadable content. Recover worksheet content?)
<img width="715" alt="screenshot 2019-02-05 16 56 11" src="https://user-images.githubusercontent.com/16313205/52286218-1a887980-2968-11e9-86a3-20ba0faa7278.png"> (translation: the file could be recovered : it is possible that some parts got deleted:
Print options
Print options
Print options
...etc: most likely one per each worksheet I cloned... 
Please help

**Comment by oleibman** [comment]
This appears to have been resolved by PR #1462 in May 2020. I have confirmed that the supplied code produces an error-free output file. Removing stale label. Leaving issue closed.
